### ISSUE TITLE
Keycloak REST API for group creation not working properly 

### ISSUE THREAD

**Issue by katariakhyati** [keycloak/keycloak#10009]
### Describe the bug

I m trying to create a new group with clients role using following API

my curl:

curl --location --request POST 'https://hostname/auth/admin/realms/master/groups'
--header 'Content-Type: application/json'
--header 'Authorization: Bearer eyJhbGciOi...'
--data-raw '{
"name": "testingRole"
"path": "/testingRole",
"attributes":

{ "test_attr": [ "true" ] }
,
"realmRoles": [],
"clientRoles":

{ "TestClient": [ "ROLE_TEST" ] }
,
"subGroups": [],
"access":

{ "view": true, "manage": true, "manageMembership": true }
}'

Once done, you will see a new group named testingRole under master realm. But if we got to role mappings section and select testingRole from Client Roles, no roles are assigned.

### Version

11.0.3 and 14.0.0

**Comment by lwohlhart** [comment]
Issue could be merged with #9354

**Comment by rmartinc** [comment]
Yes, you are right. More or less the initial idea in keycloak was that creation/update just adds the plain object (group, user or whatever) and subsequent calls are executed to establish relations between them. In your case:

1. Create the group.
2. Create the client role.
3. Call to the add client-level roles to the group: POST /{realm}/groups/{id}/role-mappings/clients/{client} in order to add the mapping.

This is more or less what the console always does. I don't think this is a major issue, moving to backlog. If anyone is interested can work on it and send a PR. But note we should think about update too, those things are not as simple as they seem.

Thanks!

**Comment by vladimir-gaydamakin** [comment]
:/

**Comment by Nosfistis** [comment]
Any updates on this?
